lib/volcanic/authenticator/v1/identity.rb

```diff
@@ -147,13 +155,14 @@ def create(name, principal_id, **opts)
         # if source is nil, default value will be taken.
         # if source is empty, it raise IdentityError
         def payload_handler(**opts)
+          source = opts[:source].to_s
           {
-            source: source = opts[:source] || 'password',
+            source: source,
             secret: opts[:secret],
             skip_secret_encryption: opts[:skip_encryption] || false,
             privileges: opts[:privilege_ids] || [],
             roles: opts[:role_ids] || [],
-            secretless: source != 'password' # secretless will be false if source is 'password'
+            secretless: opts['secretless'].nil? ? source != 'password' : opts['secretless'] # secretless will be false if source is 'password'
```

Comment:
`opts` is from kwargs. This means that the keys will all be symbols. This means that `opts['secretless']` will always be nil... `opts[:secretless]` may not be, however....

of course... your tests which ensure that when passing in a value for secretless it is respected are failing ;)

Reply:
this typo. it should accept symbol rather than string. I will fix this.